reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/client/JavaTCPClient.java

```diff
@@ -0,0 +1,119 @@
+ *  the License. You may obtain a copy of the License at
+ *  <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  <p>
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivesocket.tckdrivers.client;
+
+import io.netty.channel.Channel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.reactivesocket.ConnectionSetupPayload;
+import io.reactivesocket.DefaultReactiveSocket;
+import io.reactivesocket.DuplexConnection;
+import io.reactivesocket.ReactiveSocket;
+import io.reactivesocket.transport.tcp.TcpDuplexConnection;
+import io.reactivesocket.transport.tcp.client.TcpReactiveSocketConnector;
+import io.reactivesocket.util.Unsafe;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import rx.RxReactiveStreams;
+
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.CountDownLatch;
+
+import static java.net.InetSocketAddress.createUnresolved;
+
+// this client should parse the test cases we wrote and use them to
+public class JavaTCPClient {
+
+    public static void main(String[] args) throws MalformedURLException, URISyntaxException {
+        // we pass in our reactive socket here to the test suite
+        String file = "/Users/mjzhu/dev/reactivesocket-java/reactivesocket-tck-drivers/clienttest$.txt";
+        if (args.length > 0) {
+            file = args[0];
+        }
+        try {
+            JavaClientDriver jd = new JavaClientDriver(file);
+            jd.runTests();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static Publisher<ReactiveSocket> buildConnection(URI uri) {
+        if (uri.getScheme().equals("tcp")) {
+            ConnectionSetupPayload setupPayload = ConnectionSetupPayload.create("UTF-8", "UTF-8", ConnectionSetupPayload.NO_FLAGS);
+
+            TcpReactiveSocketConnector tcp = TcpReactiveSocketConnector.create(setupPayload, Throwable::printStackTrace);
+            Publisher<ReactiveSocket> socketPublisher = tcp.connect(new InetSocketAddress("localhost", 4567));
+            return socketPublisher;
+
+        } else {
+            throw new UnsupportedOperationException("uri unsupported: " + uri);
+        }
+    }
+
+    public static ReactiveSocket createClient() {
+        try {
+            String target = "tcp://localhost:4567/rs";
+            URI uri = new URI(target);
+
+            Publisher<ReactiveSocket> socketPublisher = buildConnection(uri);
+            SocketWrapper client = new SocketWrapper();
```

Comment:
Thanks for reviewing my code @stevegury! I used similar and also cleaner line of code from Yuri's CLI project, which has the added benefit of allowing me to add a logger to see the individual frames being sent.